app/core/src/main/java/com/fsck/k9/preferences/RealGeneralSettingsManager.kt

```diff
@@ -99,16 +100,40 @@ internal class RealGeneralSettingsManager(
         getSettings().copy(showRecentChanges = showRecentChanges).persist()
     }
 
+    override fun setAppTheme(appTheme: AppTheme) {
+        getSettings().copy(appTheme = appTheme).persist()
+    }
+
+    override fun setMessageViewTheme(subTheme: SubTheme) {
+        getSettings().copy(messageViewTheme = subTheme).persist()
+    }
+
+    override fun setMessageComposeTheme(subTheme: SubTheme) {
+        getSettings().copy(messageComposeTheme = subTheme).persist()
+    }
+
+    override fun setFixedMessageViewTheme(fixedMessageViewTheme: Boolean) {
+        getSettings().copy(fixedMessageViewTheme = fixedMessageViewTheme).persist()
+    }
```

Comment:
Please add the `@Synchronized` annotation to these methods. Otherwise it's possible to lose updates, e.g. with the following scheduling:

```
Thread 1: getSettings()
Thread 2: getSettings()
Thread 2: copy(appTheme = …)
Thread 2: persist()
Thread 1: copy(messageViewTheme = …)
Thread 1: persist()
```

In this example the `appTheme` property is overwritten with the old value again because execution of the methods wasn't atomic.